Platform/CoffeelakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -894,7 +894,7 @@ UpdatePayloadId (
     Status = GpioGetInputValue (PayloadSelGpioPad, &PayloadSelGpioData);
     if (!EFI_ERROR (Status)) {
       if (PayloadSelGpioData == 1) {
-        PayloadId = UEFI_PAYLOAD_ID_SIGNATURE;
+        PayloadId = 0;

```

Comment:
Why do you revert the logic ?
It was working,  so wondering what is the real issue.

Reply:
The old logic is level high to boot UEFI payload.  
Now you changed to level high to boot OsLoader.

Reply:
by default, the value of  "PayloadSelGpioData" is 1 when the DIP switch is OFF  and vice versa.  
 with existing code, UEFI payload is configured for both level and board always boot to UEFI payload ( this is due to changes introduced in commit 418e31ad )

Reply:
OK, I see.  Looks like some bug was introduced previously.  Make sense to me now.